src/AnalysisPrograms/Recognizers/Base/RecognizerBase.cs

```diff
@@ -292,7 +298,9 @@ public override AnalysisResult2 Analyze<T>(AnalysisSettings analysisSettings, Se
 
         public override void WriteEventsFile(FileInfo destination, IEnumerable<EventBase> results)
         {
-            Csv.WriteToCsv(destination, results.Select(x => (AcousticEvent)x));
+            // TODO check this
+            //Csv.WriteToCsv(destination, results.Select(x => (AcousticEvent)x));
+            Csv.WriteToCsv(destination, results.Select(x => (EventCommon)x));
```

Comment:
I think we should also write out the JSON variant here

Reply:
No cast will be necessary though

Reply:
Good - but I don't know how to do this.

Reply:
I've done it